packages/adminrouter/extra/src/README.md

```diff
@@ -535,8 +550,232 @@ DC/OS IAM relies on JSON Web Tokens for transmitting security information betwee
+`./test-harness/tests/test_generic.config.yml` and flavour-specific one
+from `./test-harness/tests/(open|ee)/test_generic.config.yml` are merged
+by the test code and thus the full configuration for given flavour is created.
+
+#### YAML file syntax
+Let's analyse a very simplified YAML configuration that covers only the exhibitor
+endpoint:
+```
+endpoint_tests:
+  - tests:
+      are_response_headers_ok:
+        enabled: true
+        nocaching_headers_are_sent: true
+        test_paths:
+          - /exhibitor/foo/bar
+      are_upstream_req_headers_ok:
+        enabled: true
+        jwt_should_be_forwarded: skip
+        test_paths:
+          - /exhibitor/foo/bar
+      is_upstream_correct:
+        enabled: true
+        test_paths:
+          - /exhibitor/foo/bar
+        upstream: http://127.0.0.1:8181
+      is_upstream_req_ok:
+        enabled: true
+        expected_http_ver: HTTP/1.0
+        test_paths:
+          - expected: /foo/bar
+            sent: /exhibitor/foo/bar
+          - expected: /
+            sent: /exhibitor/
+          - expected: /exhibitor/v1/cluster/status
+            sent: /exhibitor/exhibitor/v1/cluster/status
+      is_endpoint_redirecting_properly:
+        enabled: true
+        locations:
+          - path: /exhibitor
+            code: 301
+      is_unauthed_access_permitted:
+        enabled: true
+        locations:
+          - /exhibitor/exhibitor/v1/cluster/status
+      is_location_header_rewritten:
+        enabled: true
+        basepath: /exhibitor/v1/ui/index.html
+        endpoint_id: http://127.0.0.1:8181
+        redirect_testscases:
+          - location_expected: http://127.0.0.1/exhibitor/exhibitor/v1/ui/index.html
+            location_set: http://127.0.0.1/exhibitor/v1/ui/index.html
+    type:
+      - master
+      - agent
+```
+The syntax is as follows:
+* each file contains a dictionary, currently with only one key: `endpoint_tests`.
+  This is justified by the fact that we may add more generic configuration keys
+  in the future.
+* the value of the `endpoint_tests` key is a list of endpoint tests
+* lists of endpoint tests from `(open|ee)/test_generic.config.yml` and
+  `test_generic.config.yml` files are simply joined together while calculating
+  the set of tests for given AR flavour
+* the are no checks for uniqueness when it comes to endpoint tests - given
+  test can be added more than once in more than one file
+* each endpoint test is a dict which consists of:
+    * `tests`: a dict with entries describing individual tests (subtests for
+      short) of which particular endpoint test consists of
+    * `type`: a list which defines whether the subtests defined in the `tests` dict
+      can be run on `master` or `agent` or both (both keys are present in the list
+      then).
+* each subtest entry is a dictionary itself with at least one key:
+  `enabled`. If it is equal to `false` then the test is skipped/not executed.
```

Comment:
Can we make this key optional? A test should be enabled by default unless it's explicitly disabled.